1-Draft/RFCnnnn-PowerShell-Configuration.md

```diff
@@ -0,0 +1,175 @@
+- On Unix - settings files.
+
+The settings files have `Json` format.
+
+**Caution!** The settings files differ from `PowerShell Core` _profile_ files that are PowerShell scripts.
+
+Configuration schemes allow to customize `PowerShell Core` in the most flexible way:
+
+- Enterprise system administrators can use GPO, GPP and computer-wide settings files to apply approved configuration settings and mandatory security settings in a centralized manner. The same settings can be applied at user, application or startup levels.
+- Developers and consumers can use user, application and startup level settings files.
+
+### Settings locations
+
+`PowerShell Core` settings are grouped into `Policy settings` and `Regular settings`. Regular settings are normal configuration settings. Regular settings can be treated as default values. Policy settings is high priority and overlap regular settings. Policy settings are used by administrators to centrally manage applications.
+
+Location | Policy settings | Regular settings
+- | - | -
+File section | "PowerShell": { "PolicySettings": {...} } | "PowerShell": { "RegularSettings": {...} }
+File section | "OtherPowerShellApplication": { "PolicySettings": {...} } | "OtherPowerShellApplication": { "RegularSettings": {...} }
+Registry key | Software\Policies\PowerShellCore | Software\PowerShellCore
+Registry key | Software\Policies\PowerShellCore\OtherPowerShellApplication | Software\PowerShellCore\OtherPowerShellApplication
+
+### Priority of applying settings
+
+Because a configuration setting can be in several schemes, the setting wins according to the priority of its scheme.
+
+#### Priorities for Policy settings in descending order
+
+Scheme | Windows | Unix
+-| - | -
+GPO -> Computer Policy | HKLM\Software\Policies\PowerShellCore | /etc/powershell.config.json
+GPO -> User Policy | HKCU\Software\Policies\PowerShellCore | See `Comment A` below
+File -> Computer-Wide | %ProgramFiles%/PowerShell/powershell.config.json | /opt/Microsoft/powershell/powershell.config.json
+File -> User-Wide | %APPDATA%/powershell.config.json | ~/powershell.config.json
+File -> Application-Wide | $apphome/powershell.config.json | $apphome/powershell.config.json
+File -> Application-Startup | pwsh -settingsfile `somepath/powershell.config.json` | pwsh -settingsfile `somepath/powershell.config.json`
+
+#### Priorities for Regular settings in descending order
+
+Scheme | Windows | Unix
+-| - | -
+File -> Application-Startup | pwsh -settingsfile `somepath/powershell.config.json` | pwsh -settingsfile `somepath/powershell.config.json`
+File -> Application-Wide | $apphome\powershell.config.json | $apphome/powershell.config.json
+File -> User-Wide | %APPDATA%\powershell.config.json | ~/powershell.config.json
+File -> Computer-Wide | %ProgramFiles%\PowerShell\powershell.config.json | /opt/Microsoft/powershell/powershell.config.json
+GPO -> User Config | HKCU\Software\PowerShellCore | See `Comment A` below
```

Comment:
Regular setting should be deployed via GPP not GPO

Reply:
GPP is for setting the user can change.

Reply:
We're talking about values in the registry. In this case, the user can change any value if it has permissions (mainly in HKCU). (Of course then this value will be overwritten by GPO or GPP)

I don't see the point of distinguishing between regular settings and policies by _name_. What is the criterion? We are able to set the value in the registry using GPP in the Policy branch (even in HKLM) and thereby assign the policy.

The only question is _who, how and for what_ set/changes the settings. System administrators or security administrators will most often use GPOs to set mandatory standard values that are less likely to GPP to set recommended values.

This document tries to describe all the ways that can be to provide maximum flexibility and also how to get an effective value based on priority.

Reply:
GPP has the ability to not overwrite if the user changes the registry.

Reply:
Also, GPO implies that the setting is policy, which this setting is not.  It is a preference.